### PR TITLE
Fix auth check on group endpoint

### PR DESCRIPTION
**Pull request by DaoWen**
## Changes proposed in this PR

- Fix auth-check on group operations
- Simplify data-flow of context data by caching it in the `malformed?` handler
- Filter out unauthorized groups (in addition to missing groups) when using `partial` flag

## Why are we making these changes?

The authorization check for group operations was broken, meaning anyone could kill anyone else's groups.

### CODE REVIEW

**Comment by DaoWen** [comment]
@dposada - I tested this locally, but I didn't add any new integration tests. I'm looking into adding some tests using basic-auth to our integration test suite, but that will come in a separate PR.

File: scheduler/src/cook/mesos/api.clj [hunk 1]
```diff
@@ -1346,22 +1346,18 @@
...
-        [true {::guuids existing-groups}]
-        [false {::error (str "UUID "
-                             (str/join
-                               \space
-                               (set/difference (set requested-guuids) (set existing-groups)))
-                             " didn't correspond to a group")}]))
+          {existing-groups true missing-groups false} (group-by exists? requested-guuids)]
+      [false {::allow-partial-results? allow-partial-results
+              ::guuids existing-groups
+              ::bad-uuids missing-groups}])
```
Comment (dposada) [inline scheduler/src/cook/mesos/api.clj]: Should we use either `guuids` or `uuids` in both of these key names?

File: scheduler/src/cook/mesos/api.clj [hunk 2]
```diff
@@ -1371,16 +1367,26 @@
...
                                           " the following groups "
                                           (str/join \space unauthorized-guuids))}])))
-     :exists? (partial retrieve-groups conn)
+     :exists? (fn [ctx]
+                (or (::allow-partial-results? ctx)
+                    (empty? (::bad-uuids ctx))
+                    [false {::error (str "The following UUIDs didn't correspond to a group: "
+                                         (str/join
+                                           \space
+                                           (::bad-uuids ctx)))}]))
```
Comment (dposada) [inline scheduler/src/cook/mesos/api.clj]: I'm not following the logic here. Let's say `(::bad-uuids ctx)` is empty - wouldn't this cause `exists?` to return the error structure?

Comment (DaoWen) [reply scheduler/src/cook/mesos/api.clj]: The `or` here is equivalent to this:
```clojure
(cond  ; :exists?
  (::allow-partial-results? ctx) true
  (empty? (::bad-uuids ctx)) true
  :else [false {:error ...}])

File: scheduler/src/cook/mesos/api.clj (hunk 1 above)
Comment (DaoWen) [reply scheduler/src/cook/mesos/api.clj]: Renamed `::bad-uuids` → `::non-existing-guuids`.

**Pull request merged**